pkg/api/client_test.go

```diff
@@ -71,3 +70,135 @@ func TestState(t *testing.T) {
 		})
 	}
 }
+
+//----------------------------------------------
+func TestSetState(t *testing.T) {
+	for i, tc := range []struct {
+		Expected map[string]*State
+	}{
+		{
+			Expected: map[string]*State{
+				"foo": {
+					ID:  "1",
+					CPB: CPBYes,
+				},
+				"bar": {
+					ID:           "2",
+					RefBoxRole:   RoleDefenderAssist2,
+					Kinect2State: KinectStateNoBall,
+				},
+				"baz": {
+					ID:           "3",
+					ActiveDevPC:  uint8(89),
+					Temperature1: uint8(69),
+				},
+			},
+		},
+	} {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			a := assert.New(t)
+
+			out := &bytes.Buffer{}
+			in := &mockWriter{
+				WriteFunc: func(b []byte) (int, error) {
+					var m Message
+					err := json.Unmarshal(b, &m)
+					a.Nil(err)
+					a.NotEmpty(m.MessageID)
+					a.Equal(MessageTypeSetState, m.Type)
+					var ts map[string]*State
+					err = json.Unmarshal(m.Payload, &ts)
+					a.Nil(err)
+					a.Equal(ts, tc.Expected)
+
+					pld, err := json.Marshal(tc.Expected)
+					if err != nil {
+						panic(err)
+					}
+
+					err = json.NewEncoder(out).Encode(&Message{
+						MessageID: m.MessageID,
+						Type:      m.Type,
+						Payload:   pld,
+					})
+					if err != nil {
+						panic(err)
+					}
+					return len(b), nil
```

Comment:
Might want to set some boolean variable here and check it after `SetState` returns to assert that this function was actually executed.